cost/azure/databricks/rightsize_compute/azure_databricks_rightsize_compute.pt

```diff
@@ -0,0 +1,1478 @@
+  iterate $ds_virtualmachines_from_bill
+  request do
+    run_script $js_azure_databricks_managed_vm_metrics, $param_azure_endpoint, $param_stats_interval, $ds_lookback_time_formats, jq(iter_item, "[.virtualmachines[] .resourceId ]"), jq(iter_item, ".databricks_workspace.location"), jq(iter_item, ".databricks_workspace.managedResourceGroupId")
+  end
+end
+
+script "js_azure_databricks_managed_vm_metrics", type: "javascript" do
+  parameters "param_azure_endpoint", "param_stats_interval", "ds_lookback_time_formats", "resources", "location", "managedResourceGroupId"
+  result "request"
+  code <<-EOS
+  // Get the Subscription
+  subscription_id = managedResourceGroupId.split("/")[2]
+  // Lowercase the resource ids which is what getBatch API expects
+  _.each(resources, function(r, index) {
+    resources[index] = resources[index].toLowerCase()
+  })
+
+  var request = {
+    auth: "auth_azure",
+    pagination: "pagination_azure",
+    //host: param_azure_endpoint, // TODO eval param_azure_endpoint to get the hostname for metrics.monitor..
+    host: location+".metrics.monitor.azure.com",
+    verb: "POST",
+    path: "/subscriptions/"+subscription_id+"/metrics:getBatch",
+    query_params: {
+      "api-version": "2023-03-01-preview",
+      "timespan": ds_lookback_time_formats["azure_timespan"],
+      "interval": param_stats_interval,
+      "aggregation": "minimum,average,maximum",
+      "region": location,
+      "metricNamespace": "microsoft.compute/virtualmachines",
+      "metricnames": "Percentage CPU,Available Memory Bytes",
+      "orderby": "average desc"
+      // "validatedimensions": "false",
+      // "filter": "Microsoft.ResourceId eq '*'"
+      // "filter": "Microsoft.ResourceGroupName in ("+JSON.stringify(resourceids).+")"
+      // "filter": "Microsoft.ResourceId eq '*'"
+    },
+    headers: {
+      "User-Agent": "RS Policies",
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify({ "resourceids": resources })
+    // Ignore status 400, 403, and 404 which can be returned in certain (legacy) types of Azure Subscriptions
+    // ignore_status: [400, 403, 404]
+  }
+EOS
+end
+
+datasource "ds_databricks_clusters" do
+  iterate $ds_databricks_workspaces_filtered
+  request do
+    auth $auth_databricks
+    host val(iter_item, "workspaceUrl")
+    path "/api/2.0/clusters/list"
+    # Ignore 400 gets returned when using an AzureRM credential for the databricks credential ["Expected aud claim to be: 2ff814a6-3304-4ab8-85cb-cd0e6f879c1d, but was: https://management.azure.com"]
+    # Also ignore 401/403 if the Service Principal has the correct claim but just doesn't have access to the DB Workspace
+    ignore_status [400,401,403]
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "clusters[*]") do
+      field "databricks_workspace", iter_item
+      field "cluster", col_item
+    end
+  end
+end
+
+# datasource "ds_databricks_sqlwarehouses" do
```

Comment:
Defunct code should be removed unless it's here with the intent of being used at some point in the future